diff --git a/encode.go b/encode.go
index 43d5ce3..4c38a01 100644
--- a/encode.go
+++ b/encode.go
@@ -372,13 +372,10 @@ func (o *OffsetMarker) ReadOffset(buf []byte) (uint64, []byte, error) {
 	if offset > o.TotalSize {
 		return 0, nil, ErrOffset
 	}
 	if !o.HasOffset {
-		if offset != o.FixedSize {
-			return 0, nil, ErrInvalidVariableOffset
-		}
 		o.HasOffset = true
 	} else {
 		if offset < o.LastOffset {
 			return 0, nil, ErrOffsetNotIncreasing
 		}
 	}
